test/create_test.go

```diff
@@ -0,0 +1,55 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// TestCreate is unit testing for the create operation.
+func TestCreate(t *testing.T) {
+	ctx, client := newKine(t)
+
+	t.Run("CreateOne", func(t *testing.T) {
+		g := NewWithT(t)
+		resp, err := client.Txn(ctx).
+			If(clientv3.Compare(clientv3.ModRevision("testKey"), "=", 0)).
+			Then(clientv3.OpPut("testKey", "testValue")).
+			Commit()
+
+		g.Expect(err).To(BeNil())
+		g.Expect(resp.Succeeded).To(BeTrue())
+	})
+
+	t.Run("CreateExistingFails", func(t *testing.T) {
```

Comment:
It is worthy to check also for creation-after-deletion. It must be possible to insert a new that existed, but got deleted.

Reply:
Discussed offline, adding comment here for completeness. This PR is about scaffolding the testing machinery rather than adding specific unit tests. We will deal with adding more extensive tests in followup PRs.